extensions/api/data-management/asset/src/test/java/org/eclipse/dataspaceconnector/api/datamanagement/asset/service/AssetServiceImplTest.java

```diff
@@ -75,6 +79,36 @@ void query_shouldRelyOnAssetIndex() {
         assertThat(assets).hasSize(1).first().matches(hasId("assetId"));
     }
 
+    @ParameterizedTest
+    @ValueSource(strings = {
+            Asset.PROPERTY_ID,
+            Asset.PROPERTY_NAME,
+            Asset.PROPERTY_DESCRIPTION,
+            Asset.PROPERTY_VERSION,
+            Asset.PROPERTY_CONTENT_TYPE
+    })
+    void query_validFilter(String filter) {
```

Comment:
These seem like some tests to be put to the `AssetQueryValidator` class tests directly, And have a test here that mocks the validator and verifies its interaction. 
Would be the same for the other services, I have some doubt because the `subtypeMap`, but that's more a configuration than a behavior, so I'm not sure we need to test all of them, it will be enough a test one on the `QueryValidator`

Reply:
Actually, doing this by the book, I'd say we have to 
- test each service (which is already there)
- test the `QueryValidator` (to be done)
- test every variant, such as the `AssetQueryValidator` (to be done).

My argument for these (potentially duplicated) tests would be to maintain good coverage and regression stability: personally I want a test to break, if I change even the smallest thing :)